python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py

```diff
@@ -736,6 +736,13 @@ def aten〇squeeze〡shape(self: List[int]) -> List[int]:
 def aten〇squeeze〇dim〡shape(self: List[int], dim: int) -> List[int]:
     return upstream_shape_functions.squeeze(self, dim)
 
+def prims〇squeeze〡shape(a: List[int], dimensions: List[int]) -> List[int]:
```

Comment:
Can you upstream this? This op does not have the same behavior as `aten.squeeze`. In particular, if the dimension specified to squeeze is not equal to 1, the `prims` version results in an error. Also, we should be able to have the full version of this shape function, since it does not require that much more work. You just need to sort `dimensions` in decreasing order, then check the corresponding values in `a` is equal to `1` and call `upstream_shape_functions.squeeze` if it is.

Reply:
I think we can use this shape function: https://github.com/pytorch/pytorch/blob/master/torch/jit/_shape_functions.py#L429-L436, it's doing the same behaviour. It's also called the `upstream_shape_functions.squeeze` and here(https://github.com/pytorch/pytorch/blob/master/torch/jit/_shape_functions.py#L423) it checks for the shape equal to 1.

Reply:
Thanks!